azkaban-common/src/main/java/azkaban/executor/ExecutionFlowDao.java

```diff
@@ -47,9 +48,14 @@ public ExecutionFlowDao(final DatabaseOperator dbOperator) {
 
   public synchronized void uploadExecutableFlow(final ExecutableFlow flow)
```

Comment:
Not related to this change, but does this function need to be synchronized?

Reply:
I don't see why it would be necessary. It's the only synchronized method in the class so it doesn't protect any private attributes, it's only preventing concurrent calls to this method but there is nothing in there that would be a problem in those cases. I'll remove it in a separate PR, thanks.